src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -200,41 +200,38 @@ void GenericThreadStackManagerImpl_OpenThread<ImplClass>::_OnPlatformEvent(const
             }
         }
 #endif
-        Impl()->LockThreadStack();
```

Comment:
Why does the lock have to be removed? Is it safe to remove it when the Chip task can also interact with the threadInstance object?

Reply:
The function _isThreadAttached lock around the thread api to get the lock. Rest of the functionally like posting the event or the log function (that does quite a lot of process for formatting strings}
) don't need to lock the thread instance.